emscripten.py

```diff
@@ -139,11 +139,11 @@ def update_settings_glue(metadata, DEBUG):
   shared.Settings.MODULE_EXPORTS = [(asmjs_mangle(f), f) for f in metadata['exports']]
 
   if shared.Settings.STACK_OVERFLOW_CHECK:
-    if 'emscripten_stack_get_end' not in metadata['exports']:
-      logger.warning('STACK_OVERFLOW_CHECK disabled because emscripten stack helpers not exported')
-      shared.Settings.STACK_OVERFLOW_CHECK = 0
-    else:
-      shared.Settings.EXPORTED_RUNTIME_METHODS += ['writeStackCookie', 'checkStackCookie']
+    shared.Settings.EXPORTED_RUNTIME_METHODS += ['writeStackCookie', 'checkStackCookie']
+    # writeStackCookie and checkStackCookie both rely on emscripten_stack_get_end being
+    # exported.
+    if not shared.Settings.SIDE_MODULE and 'emscripten_stack_get_end' not in metadata['exports']:
+      exit_with_error('STACK_OVERFLOW_CHECK enabled by emscripten stack helpers not exported')
```

Comment:
the error text here isn't clear to me. shouldn't it be identical to the text before? if not, what is the intention of the change?

Reply:
Re-worded so it hopefully makes more sense.

The previous warning was kind of the inverse statement (it was warning about a setting we were forced to change). 
  In the new code we are instead going to error out so the message is inverted:

`STACK_OVERFLOW_CHECK disabled... `    <- this was stating the fact that we were force-ably disabling it.

`STACK_OVERFLOW_CHECK enabled but...`  <- this is giving the reason why we are erroring out.